### PR TITLE
fix(REST): RHICOMPL-2721 Fix REST API links when using v1

### PR DESCRIPTION
**Pull request by romanblanco**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: app/controllers/concerns/metadata.rb [hunk 1]
```diff
@@ -76,7 +76,9 @@ def last_offset(total)
     private
 
     def base_link_url
-      "#{path_prefix}/#{Settings.app_name}/#{controller_name}"
+      api_version = request.fullpath.delete_prefix("#{path_prefix}/#{Settings.app_name}")
+      api_version.delete_suffix!("/#{controller_name}")
```
Comment (skateman) [inline app/controllers/concerns/metadata.rb:80]: What if api version is not set at all? Like when you're trying to access the API just via `/api/compliance/...`

**Review by skateman — COMMENTED** [review]
Did you generate the openapi.json, or just rewrote its contents?

File: app/controllers/concerns/metadata.rb (hunk 1 above)
Comment (romanblanco) [reply app/controllers/concerns/metadata.rb:80]: Than `api_version` should endup being an empty string

**Comment by romanblanco** [comment]
> Did you generate the openapi.json, or just rewrote its contents?

@skateman I did not generate them, I used `sed` to update the paths.

**Pull request merged**